tools/site_admin/manage_special_days.php

```diff
@@ -33,16 +33,21 @@
 
         $spec_code = trim($_POST['spec_code']);
 
-        $source = new SpecialDay($spec_code);
+        if (strlen($spec_code) < 1) {
+            $errmsgs .= _("A value for Special Day ID is required. Please enter one.") . "<br>";
+        } elseif (preg_match("/[^A-Za-z0-9. -]/", $spec_code) === 1) {
```

Comment:
We only want to do this validation for new entries, not for existing entries. This is most important for other sites which might have already created some special days with these characters in them.

Reply:
Or perhaps leave out this check.